uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-assignment_test.js

```diff
@@ -0,0 +1,104 @@
+define([
+    'common/angular',
+    'mocks/assignment',
+    'common/angularMocks',
+    'tasks-assignments/app'
+], function (angular, Mock) {
+    'use strict';
+
+    describe('ModalAssignmentCtrl', function () {
+        var ctrl, modalInstance, scope, AssignmentService, TaskService, DocumentService, ContactService, HR_settings;
+
+        beforeEach(module('civitasks.appDashboard'));
+        beforeEach(inject(function ($controller, $rootScope) {
+            modalInstance = {
+                close: jasmine.createSpy('modalInstance.close'),
+                dismiss: jasmine.createSpy('modalInstance.dismiss'),
+                result: {
+                    then: jasmine.createSpy('modalInstance.result.then')
+                }
+            };
+
+            scope = $rootScope.$new();
+
+            AssignmentService = {};
+            TaskService = {};
+            DocumentService = {};
+            ContactService = {};
+            HR_settings = {
+                DATE_FORMAT: 'DD/MM/YYYY'
+            };
+
+            ctrl = $controller('ModalAssignmentCtrl', {
+                $scope: scope,
+                $modalInstance: modalInstance,
+                AssignmentService: AssignmentService,
+                TaskService: TaskService,
+                DocumentService: DocumentService,
+                ContactService: ContactService,
+                data: {},
+                config: {},
+                settings: {},
+                HR_settings: HR_settings
+            });
+        }));
+
+        it('Controller should be defined', function () {
+            expect(ctrl).toBeDefined()
+        });
+
+        describe('Bulk Assign', function(){
+            beforeEach(function(){
+                angular.extend(scope, Mock);
+            });
+
+            it('Test copyAssignee method', function(){
```

Comment:
Please write tests in a more readable, english-like way. For example:
```js
describe('copyAssignee()', function () {
    var list, assigneeId = [2];

    beforeEach(function () {
        list = scope.taskList;
        list[0].assignee_contact_id = assigneeId;

        scope.copyAssignee(list);
    });

    it("assigns the assignee's id of the first item to the whole list", function () {
        expect(list.every(function (item) {
            return item.assignee_contact_id.toEqual(assigneeId);
        })).toBe(true);
    });
});
```
_(I don't think there is the necessity here to test the "before" status, it's pretty unlikely that all the items have already the same assignee id, and in any case you are the one who wrote the mocked list, and thus the initial status of it)_

The goal here is to have a test suite for components that you can quickly read through (by just focusing on the `describe` and `it` description) and get a good idea of what they do, what their interface is, etc. A simple `Test copyAssignee method` doesn't really give any information of any kind to the reader